src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/ProcessCompletionOfNextDataSetVersionImportFunctionTests.cs

```diff
@@ -117,6 +124,3168 @@ private static Mock<TaskOrchestrationContext> DefaultMockOrchestrationContext(Gu
+                                        .SetPublicId(SqidEncoder.Encode(8)))
+                                .Generate(4)))
+                .GenerateList(2);
+
+            var (originalVersion, _) = await CreateDataSetInitialVersion(oldFilterMeta);
+
+            var newFilterMeta = DataFixture.DefaultFilterMeta()
+                .ForIndex(0, UnchangedFilterMetaSetter(
+                    filterMeta: oldFilterMeta[0],
+                    newOptionLinks: () => DataFixture.DefaultFilterOptionMetaLink()
+                        .ForIndex(0, s =>
+                            s.SetOption(DataFixture.DefaultFilterOptionMeta())
+                                .SetPublicId(SqidEncoder.Encode(3))) // Filter Option changed
+                        .ForIndex(1, s =>
+                            s.SetOption(DataFixture.DefaultFilterOptionMeta())
+                                .SetPublicId(SqidEncoder.Encode(4))) // Filter Option changed
+                        .ForIndex(2, s =>
+                            s.SetOption(
+                                    DataFixture.DefaultFilterOptionMeta()
+                                        .WithLabel("j"))
+                                .SetPublicId(SqidEncoder.Encode(9))) // Filter Option added
+                        .ForIndex(3, s =>
+                            s.SetOption(
+                                    DataFixture.DefaultFilterOptionMeta()
+                                        .WithLabel("e"))
+                                .SetPublicId(SqidEncoder.Encode(10))) // Filter Option added
+                        .Generate(4)))
+                .ForIndex(1, UnchangedFilterMetaSetter(
+                    filterMeta: oldFilterMeta[1],
+                    newOptionLinks: () => DataFixture.DefaultFilterOptionMetaLink()
+                        .ForIndex(0, s =>
+                            s.SetOption(DataFixture.DefaultFilterOptionMeta())
+                                .SetPublicId(SqidEncoder.Encode(7))) // Filter Option changed
+                        .ForIndex(1, s =>
+                            s.SetOption(DataFixture.DefaultFilterOptionMeta())
+                                .SetPublicId(SqidEncoder.Encode(8))) // Filter Option changed
+                        .ForIndex(2, s =>
+                            s.SetOption(
+                                    DataFixture.DefaultFilterOptionMeta()
+                                        .WithLabel("g"))
+                                .SetPublicId(SqidEncoder.Encode(11))) // Filter Option added
+                        .ForIndex(3, s =>
+                            s.SetOption(
+                                    DataFixture.DefaultFilterOptionMeta()
+                                        .WithLabel("l"))
+                                .SetPublicId(SqidEncoder.Encode(12))) // Filter Option added
+                        .Generate(4)))
+                .GenerateList(2);
```

Comment:
But aren't we testing across multiple filters in the ordering test? It seems like we're testing this pretty comprehensively there so having all this extra test data across many test cases just seems unnecessarily redundant.

This is potentially adding an extra 500 - 1000 lines of test code that we could trim away! I'd personally prioritise readability here.

Reply:
I think we'd have to jump on a call about this one. I'm not sure there are as many redundant lines of code as you may suggest. There are possibly some somewhere, but I'm currently of the impression that **most** of the test set-up code has some value. For example, I'd treat additions / deletions / updates all separately - so even if I'm testing across multiple filters in one of those tests, say just for additions, I'd still want to do it for the others (updates and deletions) as it is testing something slightly different. I.e. I'd be testing, separately:
- Additions work across multiple filters
- Deletions work across multiple filters
- Updates work across multiple filters

I'd also still want to test across multiple filters in the ordering tests because this, again, is testing that the ordering works across multiple filters, which is slightly different again.